globus_cli/commands/endpoint/update.py

```diff
@@ -15,11 +15,43 @@
 def endpoint_update(endpoint_id, display_name, description, organization,
                     department, keywords, contact_email, contact_info,
                     info_link, public, default_directory, force_encryption,
-                    oauth_server, myproxy_server, myproxy_dn):
+                    oauth_server, myproxy_server, myproxy_dn, network_use,
+                    custom_concurrency, custom_parallelism, location_automatic,
+                    location, disable_verify):
     """
     Executor for `globus endpoint update`
     """
-    client = get_client()
+    # require custom_concurrency and custom_parallelism for custom network_use
+    if network_use == "custom" and (not custom_concurrency or
+                                    not custom_parallelism):
+        raise click.UsageError(
+            "custom network_use level requires --custom-concurrency and "
+            "--custom-parallelism to be set")
+
+    # set max and preferred concurrency and parallelism based on args
+    if custom_concurrency:
+        max_concurrency, preferred_concurrency = custom_concurrency
+    else:
+        max_concurrency = None
+        preferred_concurrency = None
+    if custom_parallelism:
+        max_parallelism, preferred_parallelism = custom_parallelism
+    else:
+        max_parallelism = None
+        preferred_parallelism = None
+
+    # location cannot be automatic and user defined
+    if location_automatic and location:
+        raise click.UsageError(
+            "cannot combine --location and --location-automatic")
+
+    # set location based on args
+    if location_automatic:
+        location = "Automatic"
+    elif location:
+        location = "{},{}".format(location[0], location[1])
+    else:
+        location = None
```

Comment:
I'm not super-ecstatic about repeating all of this logic. Can we dump a nice little helper somewhere to handle post-parsing validation/transformation on all of these options?

This is the only issue I have with this PR -- everything else seems sane/fine.